api/src/shared/infrastructure/utils/promise-utils.js

```diff
@@ -41,4 +41,23 @@ function map(array, mapper, { concurrency = Infinity } = {}) {
   });
 }
 
-export const PromiseUtils = { map };
+/**
+ * Asynchronously maps in serie over an array of values and keeps the array order.
+ *
+ * @param {Array} array - The array of values to be mapped.
+ * @param {Function} mapper - A function that takes a value and its index, and returns a promise or value.
+ *
+ * @returns {Promise<Array>} A promise that resolves to an array of mapped results.
+ *
+ */
+async function mapSeries(array, mapper) {
+  const result = [];
+  for (let index = 0; index < array.length; index++) {
+    const item = array[index];
```

Comment:
```suggestion
 for (const [index, item] of array.entries()) {
```
c'est kif kif bourricot, j'ai juste une aversion naturelle pour les boucles for.